pkg/events/events.go

```diff
@@ -0,0 +1,50 @@
+package events
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+)
+
+// Event message format "medik8s <operator shortname> <message>"
+const customFmt = "[remediation] %s"
+
+// NormalEventf will record an event with type Normal and fixed message.
+func NormalEvent(recorder record.EventRecorder, object runtime.Object, reason, message string) {
+	recorder.Event(object, "Normal", reason, fmt.Sprintf("[remediation] %s", message))
+}
+
+// NormalEventf will record an event with type Normal and formatted message.
+func NormalEventf(recorder record.EventRecorder, object runtime.Object, reason, messageFmt string, a ...interface{}) {
+	message := fmt.Sprintf(messageFmt, a...)
+	recorder.Event(object, "Normal", reason, fmt.Sprintf("[remediation] %s", message))
+}
+
+// WarningEventf will record an event with type Warning and fixed message.
+func WarningEvent(recorder record.EventRecorder, object runtime.Object, reason, message string) {
+	recorder.Event(object, "Warning", reason, fmt.Sprintf("[remediation] %s", message))
+}
+
+// WarningEventf will record an event with type Warning and formatted message.
+func WarningEventf(recorder record.EventRecorder, object runtime.Object, reason, messageFmt string, a ...interface{}) {
+	message := fmt.Sprintf(messageFmt, a...)
+	recorder.Event(object, "Warning", reason, fmt.Sprintf("[remediation] %s", message))
+}
+
+// Special case events
+
+// RemediationStarted will record a Normal event with reason RemediationStarted and message Remediation started.
+func RemediationStarted(recorder record.EventRecorder, object runtime.Object) {
+	NormalEvent(recorder, object, "RemediationStarted", "Remediation started")
+}
+
+// RemediationStoppedByNHC will record a Normal event with reason RemediationStopped.
+func RemediationStoppedByNHC(recorder record.EventRecorder, object runtime.Object) {
+	NormalEvent(recorder, object, "RemediationStopped", "NHC added the timed-out annotation, remediation will be stopped")
```

Comment:
I'd rather keep it this way. I understand creates a bit of frictions writing the tests, but I think it's a good friction. It would improve test readability (you actually read in the test source code what text is expected, without looking at the variable's value) and would spot typos easily.

Reply:
I have no strong preference, fine by me.